tests/pytorch/nightly/dlrm.libsonnet

```diff
@@ -82,94 +88,122 @@ local utils = import 'templates/utils.libsonnet';
-          --tpu-cores=8
-      ||| % command_common
+        pip install onnx
+        git clone --recursive https://github.com/pytorch-tpu/examples.git
+        %s 
+      ||| % utils.toCommandString(self.paramsOverride.trainCommand),
     ),
   },
   local mp_dp_fwd = common.Functional {
     modelName: 'dlrm-mpdp-fwd',
+    paramsOverride+:: {
+      miniBatchSize: 2048,
+      archEmbeddingSize: '1000000-1000000-1000000-1000000-1000000-1000000-1000000-1000000',
+      tpuModelParallelGroupLen: 4,
+      tpuCores: 8,
+    },
     command: utils.scriptCommand(
       |||
-        %(command_common)s  --mini-batch-size=2048 \
-          --arch-embedding-size=1000000-1000000-1000000-1000000-1000000-1000000-1000000-1000000 \
-          --tpu-model-parallel-group-len 4 \
-          --tpu-cores=8
-      ||| % command_common
+        pip install onnx
+        git clone --recursive https://github.com/pytorch-tpu/examples.git
+        %s 
+      ||| % utils.toCommandString(self.paramsOverride.trainCommand),
     ),
   },
-  local convergence_common = |||
-    --raw-data-file=/datasets/criteo-kaggle-mm/train.txt \
-    --processed-data-file=/datasets/criteo-kaggle-mm/kaggleAdDisplayChallenge_processed.npz \
-    --memory-map \
-    --arch-sparse-feature-size=16 \
-    --arch-mlp-bot="13-512-256-64-16" \
-    --arch-mlp-top="512-256-1" \
-    --data-generation=dataset \
-    --data-set=kaggle \
-    --loss-function=bce \
-    --round-targets=True \
-    --learning-rate=0.1 \
-    --mini-batch-size=128 \
-    --print-freq=1024 \
-    --print-time \
-    --test-mini-batch-size=16384 \
-    --test-freq=101376 \
-    --use-tpu \
-    --num-indices-per-lookup=1 \
-    --num-indices-per-lookup-fixed \
-    --tpu-model-parallel-group-len 8 \
-    --tpu-metrics-debug \
-    --tpu-cores=8 |& tee dlrm_logs.txt
-    acc=`grep Testing dlrm_logs.txt | tail -1 | grep -oP 'best \K[+-]?([0-9]*[.])?[0-9]+'`
-    echo 'Accuracy is' $acc
-    test $(echo $acc'>'78.75 | bc -l) -eq 1  # assert cls acc higher than 78.75
-  |||,
+
   local criteo_kaggle = common.Convergence {
+    paramsOverride+:: {
+      tpuCores: 8,
+      miniBatchSize: 128,
+      tpuModelParallelGroupLen: 8,
+      ArchSparseFeatureSize: 16,
+      ArchMlpBot: '13-512-256-64-16',
+      ArchAlpTop: '512-256-1',
+      NumIndicesPerLookup: 1,
+      trainCommand+: [
+        '--raw-data-file=/datasets/criteo-kaggle-mm/train.txt',
+        '--processed-data-file=/datasets/criteo-kaggle-mm/kaggleAdDisplayChallenge_processed.npz',
+        '--memory-map',
+        '--data-generation=dataset',
+        '--print-time',
+        '--test-mini-batch-size=16384',
+        '--test-freq=101376',
+        '--data-set=kaggle',
+        '--loss-function=bce',
+        '--round-targets=True',
+        '--learning-rate=0.1',
+        '--print-freq=1024',
+        '--no-save',
+        '--max-epoch=1',
+      ],
+    },
     command: utils.scriptCommand(
       |||
+        set +e
         apt-get install -y bc
```

Comment:
Can you include these setup steps in `tpuVmExtraSetup` instead? Example:

https://github.com/GoogleCloudPlatform/ml-testing-accelerators/blob/fef030dc06ecb685639230f05e2c317add1cb563/tests/pytorch/nightly/fs-transformer.libsonnet#L188-L196

Reply:
test dlrm tests are all 2vm tests, I can definitely move the four lines into a separate definition, but it won't be used elsewhere except this single case here. I'm keeping it as is if no other reasons.

Reply:
I see, I missed that these were 2vm. You shouldn't need to clone the PyTorch examples for 2vm because this repository is already included in the PyTorch XLA docker image: https://github.com/pytorch/xla/blob/e8353135e0e057df495653a7bc8ac2c00ce60b82/docker/Dockerfile#L49